src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -185,6 +190,15 @@ public Assembly LoadFromAssemblyName(AssemblyName assemblyName)
             Assembly assembly = Load(assemblyName);
             if (assembly == null)
             {
+                // Invoke the AssemblyResolve event callbacks if wired up
+                if (AssemblyResolve != null)
```

Comment:
Thread safety - I think you need to cache it in local variable, and do both the null check and call against the same value.